docs/syntax/Cadence.syntax

```diff
@@ -0,0 +1,1482 @@
+        returnKeyword expression
+    ;
+
+returnStatementEmpty:
+        returnKeyword
+    ;
+
+breakStatement:
+        breakKeyword
+    ;
+
+continueStatement:
+        continueKeyword
+    ;
+
+ifStatement:
+        ifKeyword ifStatementCondition block *1ifStatementElse
+    ;
+
+ifStatementCondition:
+        expression
+    |   variableDeclaration
+    ;
+
+ifStatementElse:
+        elseKeyword ifStatementElseElement
+    ;
+
+ifStatementElseElement:
+        ifStatement
+    |   block
+    ;
+
+whileStatement:
+        whileKeyword expression block
+    ;
+
+forStatement:
+        forKeyword *1forStatementIndex nonReservedIdentifier inKeyword expression block
+    ;
+
+forStatementIndex:
+        nonReservedIdentifier ","
+    ;
+
+switchStatement:
+        switchKeyword expression "{" *switchCase "}"
+    ;
+
+switchCase:
+        switchCaseExpression
+    |   switchCaseDefault
+    ;
+
+switchCaseExpression:
+        caseKeyword expression ":" statements
+    ;
+
+switchCaseDefault:
+        defaultKeyword ":" statements
+    ;
+
+emitStatement:
+        emitKeyword 1*identifier( "." ) *1invocationTemplateParameters "(" *1argumentList ")"
+    ;
+
+variableDeclaration:
+        letVariableDeclaration
+    |   varVariableDeclaration
+    ;
+
+letVariableDeclaration:
+        access letKeyword nonReservedIdentifier *1variableDeclarationTypeAnnotation transfer expression *1variableDeclarationTrailingTransferExpression
+    ;
+
+varVariableDeclaration:
+        access varKeyword nonReservedIdentifier *1variableDeclarationTypeAnnotation transfer expression *1variableDeclarationTrailingTransferExpression
+    ;
+
+variableDeclarationTypeAnnotation:
+        ":" typeAnnotation
+    ;
+
+variableDeclarationTrailingTransferExpression:
+        transfer expression
+    ;
+
+assignment:
+        assignmentTarget transfer assignmentValue
+    ;
+
+assignmentTarget:
+        expression
+    ;
+
+assignmentValue:
+        expression
+    ;
+
+swap:
+        swapExpressionLeft "<->" swapExpressionRight
+    ;
+
+swapExpressionLeft:
+        expression
+    ;
+
+swapExpressionRight:
+        expression
+    ;
+
+transfer:
+        transferEqual
+    |   transferMoveForced
+    |   transferMove
+    ;
+
+transferEqual:
+        "="
+    ;
+
+transferMoveForced:
+        moveForced
+    ;
+
+transferMove:
+        move
+    ;
+
+expression:
+        conditionalExpression  // 20
+    ;
+
+conditionalExpression:
+        orExpression *1conditionalExpressionBranches
+    ;
+
+conditionalExpressionBranches:
+        "?" conditionalExpressionBranchesTrue ":" conditionalExpressionBranchesFalse
+    ;
+
+conditionalExpressionBranchesTrue:
+        expression
+    ;
+
+conditionalExpressionBranchesFalse:
+        expression
+    ;
+
+orExpression:  // 30 '||'
+        1*andExpression( "||" )
+    ;
+
+andExpression:  // 40
+        1*comparisonExpression( "&&" )
+    ;
+
+comparisonExpression:  // 50
+        1*2nilCoalescingExpression( comparisonOp )
```

Comment:
Though type checking might reject nested comparison expressions, parsing still allows them.
So this should not have an upper bound in the multiplicity:
```suggestion
        1*nilCoalescingExpression( comparisonOp )
```